### PR TITLE
add handshake logic in tracker

### PR DESCRIPTION
**Pull request by air-upc**
Add handshake logic when client lunchs the first request to server.

### CODE REVIEW

**Comment by wooparadog** [comment]
LGTM.

File: thriftpy/contrib/tracking/tracking.thrift [hunk 1]
```diff
@@ -11,4 +11,6 @@ struct RequestHeader {
  * This is the struct that a successful upgrade will reply with.
  */
 struct UpgradeReply {}
-struct UpgradeArgs {}
+struct UpgradeArgs {
+    1: string app_id
```
Comment (wkingfly) [inline thriftpy/contrib/tracking/tracking.thrift]: better named `source_id`

Comment (wooparadog) [inline thriftpy/contrib/tracking/tracking.thrift]: Let's just directly export our naming :P

**Comment by wooparadog** [comment]
@air-upc ci failed

**Comment by wooparadog** [comment]
@lxyu doesn't seem to be related to this pr, take a look?

**Comment by air-upc** [comment]
@wooparadog three jobs failed after this pr submited at first, now only one left; hard to say what happend.

**Comment by lxyu** [comment]
The failure is because of ports conflict in travic-ci machine, it usually works after some retries.

I triggered re-run for the failed test, 2 succeed and the last faild test can't be triggered to re-run. Seems a travis-ci bug.

**Comment by wooparadog** [comment]
@lxyu if there isn't any problem, can we merge this request and push a new version?

**Comment by lxyu** [comment]
yah, merged and triggered a new ci job.

**Comment by wkingfly** [comment]
话说这个pr是在 0.3.3.1 里面吗，看tags里没有……

**Comment by wooparadog** [comment]
是应该在这个里面.

**Pull request merged**